### PR TITLE
Internal: PHPUnit - Tests throw warning messages [ED-8311]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: tests/bootstrap.php [hunk 1]
```diff
@@ -70,6 +70,14 @@
 	$experiments_manager->set_feature_default_state( $experimental_data['name'], $experiments_manager::STATE_ACTIVE );
 }, 10, 2 );
 
+/**
+ * Override shutdown function of the logger, to prevent logging errors during tests.
+ * core/logger/manager.php:266
```
Comment (StyleShit) [inline tests/bootstrap.php:75]:
```suggestion
 * https://github.com/elementor/elementor/blob/70a0a4dec2fa5575b0339f21af4f25ebff3c2421/core/logger/manager.php#L266
```

File: tests/bootstrap.php [hunk 2]
```diff
@@ -70,6 +70,14 @@
...
+ */
+register_shutdown_function( function () {
+	exit();
```
Comment (StyleShit) [inline tests/bootstrap.php:78]:
```suggestion
	die;
```

we don't have a single `exit()` in our codebase, so it looks weird 😅

File: tests/phpunit/elementor/includes/managers/test-image.php [hunk 3]
```diff
@@ -101,6 +73,31 @@ public function test_delete_custom_images() {
 
 		// Cleanup
 		unlink( $base_full_path );
-		$this->delete_image( $attachment_id );
+	}
+
+	private function create_image() {
```
Comment (StyleShit) [inline tests/phpunit/elementor/includes/managers/test-image.php:78]: move to top the reduce diff

File: tests/phpunit/elementor/modules/dev-tools/test-deprecation.php [hunk 4]
```diff
@@ -226,8 +226,12 @@ public function test_do_deprecated_action__soft() {
 			echo 'Testing Do Deprecated Action';
 		} );
 
+		ob_start();
```
Comment (StyleShit) [inline tests/phpunit/elementor/modules/dev-tools/test-deprecation.php:229]: maybe add a comment? (well, actually the next dev will see the commit message, so I think it's ok?)

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Comment by StyleShit** [comment]
Maybe change to internal

File: tests/bootstrap.php (hunk 1 above)
Comment (Nevoss) [reply tests/bootstrap.php]: Removed

File: tests/bootstrap.php (hunk 2 above)
Comment (Nevoss) [reply tests/bootstrap.php]: Removed

File: tests/phpunit/elementor/includes/managers/test-image.php (hunk 3 above)
Comment (Nevoss) [reply tests/phpunit/elementor/includes/managers/test-image.php:78]: Done

File: tests/phpunit/elementor/modules/dev-tools/test-deprecation.php (hunk 4 above)
Comment (Nevoss) [reply tests/phpunit/elementor/modules/dev-tools/test-deprecation.php:229]: Why comment?

**Review by StyleShit — APPROVED** [review]

File: tests/phpunit/elementor/includes/managers/test-image.php [hunk 5]
```diff
@@ -30,10 +35,6 @@ public function create_image() {
 		return $attachment_id;
 	}
 
-	public function delete_image( $attachment_id ) {
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/managers/test-image.php:33]: Why? We don't need to clean up?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/includes/managers/test-image.php (hunk 5 above)
Comment (Nevoss) [reply tests/phpunit/elementor/includes/managers/test-image.php:33]: WordPress cleans the database for each test.
and for the image itself, there is another method that does it.

**Pull request merged**